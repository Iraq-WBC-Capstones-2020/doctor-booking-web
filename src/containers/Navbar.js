import React from 'react';
import { Navbar, Nav, Container } from 'react-bootstrap';
function nav() {
  return (
    <Container>
      <Navbar collapseOnSelect expand="lg" bg="white" variant="light">
        <Navbar.Brand>Doctor Booking</Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="ml-auto justify-content-end">
            <Nav.Link>Home</Nav.Link>
            <Nav.Link>Search</Nav.Link>
            <Nav.Link>About</Nav.Link>
            <Nav.Link>FAQ</Nav.Link>
            <Nav.Link>Sign in</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </Container>
  );
}

export default nav;
