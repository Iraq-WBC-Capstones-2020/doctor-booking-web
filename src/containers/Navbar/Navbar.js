import React, { useState, useContext, useEffect } from 'react';
import { Navbar, Nav, Container, NavDropdown } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import { auth } from '../../Firebase';
import { ACTIONS, DoctorContext } from '../../DoctorContext';
import { firebaseFunctions } from '../../firebaseFunctions';
function NavBar() {
  const [state, dispatch] = useContext(DoctorContext);
  const [language, setLanguage] = useState('English');
  const changeLanguage = (lng) => {
    setLanguage(lng);
  };
  const handleSignOut = () => {
    firebaseFunctions.signOut();
    dispatch({ type: ACTIONS.IS_SIGNED_IN, isSignedIn: false });
  };
  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      if (user) {
        console.log(user);
        dispatch({ type: ACTIONS.IS_SIGNED_IN, isSignedIn: true });
      }
    });
  }, []);
  return (
    <div>
      <Navbar
        className="shadow-sm"
        collapseOnSelect
        expand="lg"
        bg="white"
        variant="light"
      >
        <Container>
          <NavLink to="/" exact>
            <Navbar.Brand>DOCTOR BOOKING</Navbar.Brand>
          </NavLink>
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse id="responsive-navbar-nav">
            <Nav className="ml-auto justify-content-end">
              <NavLink
                activeClassName="active-link"
                className="nav-link"
                exact
                to="/"
              >
                Home
              </NavLink>
              <NavLink
                activeClassName="active-link"
                className="nav-link"
                to="/search"
              >
                Search
              </NavLink>
              <NavLink
                activeClassName="active-link"
                className="nav-link"
                to="/about"
              >
                About
              </NavLink>
              <NavLink
                activeClassName="active-link"
                className="nav-link"
                to="/faq"
              >
                FAQ
              </NavLink>
              {state.isSignedIn ? (
                <a
                  activeClassName="active-link"
                  className="nav-link"
                  onClick={handleSignOut}
                  href="/"
                >
                  Sign Out
                </a>
              ) : (
                <NavLink
                  activeClassName="active-link"
                  className="nav-link"
                  to="/signin"
                >
                  Sign in
                </NavLink>
              )}
              <NavDropdown title={language}>
                <NavDropdown.Item onClick={() => changeLanguage('English')}>
                  English
                </NavDropdown.Item>
                <NavDropdown.Item onClick={() => changeLanguage('كوردى')}>
                  كوردى
                </NavDropdown.Item>
                <NavDropdown.Item onClick={() => changeLanguage('عربي')}>
                  عربي
                </NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </div>
  );
}

export default NavBar;
