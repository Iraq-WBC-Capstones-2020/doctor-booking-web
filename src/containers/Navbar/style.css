.mainNav {
  z-index: 10000;
}

nav a {
  color: #026fc2 !important;
}
/*# sourceMappingURL=style.css.map */
