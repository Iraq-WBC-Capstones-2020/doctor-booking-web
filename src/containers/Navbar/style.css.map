{
    "version": 3,
    "mappings": "AACA,OAAO,CAAC,2EAAI;AACZ,AAAA,YAAY,CAAC;EACX,KAAK,ECHQ,OAAO,CDGC,UAAU;CAChC;;AAED,AAAA,OAAO,CAAC;EACN,UAAU,EAAE,IAAI;CAQjB;;AATD,AAEE,OAFK,CAEL,SAAS,CAAC;EACR,SAAS,EAAE,IAAI;CAChB;;AAJH,AAKE,OALK,CAKL,aAAa,CAAC;EACZ,WAAW,EAAE,cAAc;EAC3B,SAAS,EAAE,IAAI;CAChB;;AAGH,AAAA,aAAa,CAAC;EACZ,OAAO,EAAE,MAAM;EACf,MAAM,EAAE,IAAI;EACZ,WAAW,EAAE,IAAI;CAClB;;AAED,AAAA,cAAc,CAAC;EACb,4CAA4C;EAC5C,UAAU,EAAE,IAAI;EAChB,OAAO,EAAE,mBAAmB;CAC7B;;AAED,MAAM,EAAE,SAAS,EAAE,KAAK;EACtB,AAAA,WAAW,GAAG,EAAE,GAAG,CAAC,CAAC;IACnB,+CAA+C;IAC/C,WAAW,EAAE,MAAM;IACnB,cAAc,EAAE,MAAM;IACtB,WAAW,EAAE,IAAI;GAClB",
    "sources": [
        "style.scss",
        "../../styles/_variables.scss"
    ],
    "names": [],
    "file": "style.css"
}