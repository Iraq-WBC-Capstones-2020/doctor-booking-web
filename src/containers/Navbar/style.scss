@import '../../styles/variables';

.mainNav {
  z-index: 10000;
}

nav {
  a {
    color: $primaryColor !important;
  }
}
