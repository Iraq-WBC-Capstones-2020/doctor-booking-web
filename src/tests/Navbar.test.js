import React, { Suspense } from 'react';
import Navbar from '../containers/Navbar/Navbar';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import DoctorInfoProvider from '../DoctorContext';

test('navbar renders correctly', () => {
  const tree = renderer
    .create(
      <DoctorInfoProvider>
        <MemoryRouter>
          <Suspense fallback={<div>loading...</div>}>
            <Navbar />
          </Suspense>
        </MemoryRouter>
      </DoctorInfoProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
