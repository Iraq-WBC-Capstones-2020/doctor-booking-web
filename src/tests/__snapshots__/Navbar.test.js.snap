// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders correctly 1`] = `
<div>
  <nav
    className="shadow-sm navbar navbar-expand-lg navbar-light bg-white"
  >
    <div
      className="container"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="navbar-brand"
        >
          Doctor Booking
        </span>
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="ml-auto justify-content-end navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            aria-current="page"
            className="nav-link active-link"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            navBar.links.home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/search"
            onClick={[Function]}
          >
            navBar.links.search
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            navBar.links.about
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/faq"
            onClick={[Function]}
          >
            navBar.links.FAQ
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/signin"
            onClick={[Function]}
          >
            signIn
          </a>
          <div
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            />
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
